apps/api/src/report/report.util.service.ts

```diff
@@ -120,44 +120,113 @@ export class ReportUtilService {
     `;
   }
 
+  /*
+  Report 3
+  SQL query explains:
+  Here we are counting active, hired, and withdrawn application TO date.
+
+  withdrawn_applicants:
+    First, we will list down all the applicants who withdraw from the process(till TO date).
+  reactive_applicants:
+    Based on the withdrawal list, Let's try to get the latest milestone other than withdrawal to mark as re-active.
+  hired_applicants:
+    Let's find hired applicants based on the milestone 28.
+    If a candidate withdraws any time during the process, in this case, the hire date must be greater than the withdrawal date.
+  applicants:
+    Let's add hired, withdrawn, or reactive to each applicant row.
+  report:
+    Few conditions that need to check to identify such value(hired, withdrawal, and re-active).
+    Like hired applicants are one who may get withdrawn but later re-active, or not withdrawn from the process at all.
+
+  UNION ALL
+    We have to identify two kinds of records stat,
+    One before {from} date and One after {from} date. that's why UNION ALL.
+  */
   hiredActiveWithdrawnApplicantCountQuery(from: string, to: string) {
     return `
       -- 28 hired
       -- 20 Withdrew
-      WITH old_applicants AS (
-        SELECT 
-          applicants.id,
-          applicants.registration_date,
-          CASE WHEN COALESCE(ien_staus_hired.status_id, 0) > 0 THEN 1 ELSE 0 END as hired,
-          CASE WHEN COALESCE(ien_staus_withdrew.status_id, 0) > 0 THEN 1 ELSE 0 END as withdrawn
+      WITH withdrawn_applicants AS (
+        SELECT applicant_id, max(start_date) as start_date
+        FROM public.ien_applicant_status_audit 
+        WHERE status_id=20 
+          AND start_date::date <= '${to}'
+        GROUP BY applicant_id
+      ), reactive_applicants AS (
+        SELECT wa.applicant_id, min(ien.start_date) as start_date
+        FROM withdrawn_applicants wa INNER JOIN public.ien_applicant_status_audit ien ON wa.applicant_id=ien.applicant_id
+        WHERE ien.start_date::date >= wa.start_date
+          AND ien.start_date::date <= '${to}'
+          AND ien.status_id != 20
```

Comment:
Is the reason that it has status withdrew(20) even though it has later milestone than 20,  because status comes from ATS?

Reply:
@sunghwan-park-fw 
Yes, applicants can restart application too even after withdrawal.
If applicant is not responsive for longer period then it will be withdrawn but later applicant came again and want to start from where it left.

Reply:
Also applicant can withdraw from recruitment or immigration stage too. Or hold application by withdrawing it and start again.